### DISCUSSION TITLE
[Help] IF Condition based on Left Sidebar OPEN/CLOSE

### DISCUSSION THREAD

**Discussion by FelipeRearden** [SilentVoid13/Templater#1049] [Help] [answered]
Hello friends !!!!!

I wanna built a templater script that executes commands based on the **state** of the **Left Sidebar**
→ `this.app.workspace.leftSplit.expand`
→ `this.app.workspace.leftSplit.collapse`

In a nutshell what I need to is something like this 

```
if (this.app.workspace.leftSplit.expand) {
 window.open("URL_B");
} else if (this.app.workspace.leftSplit.collapse) {
 window.open("URL_A");
}
```

ℹ️ Where `URL_A` and `URL_B` are commands generated with **Advanced URI plugin**

### My Issue :(

I don't know how to define the **if statement** to indicate that the Sidebar is Open or Closed :(

I hope someone could give some help 🙏

---

Thanks for reading this !

Have a great day 🌞

**Comment by welpdx** [comment, marked as answer]
Hey, 

Here are some explanation of functions as you already know:

`this.app.workspace.leftSplit.expand()` -> expands the left side bar

`this.app.workspace.leftSplit.collapse()` -> hides the left side bar

`this.app.workspace.leftSplit.collapsed` -> if collapsed, it returns `true`. if expanded, it will return `false`

---

Are you executing a templater script via Open Insert Template modal, which will then check the left side bar's condition? if so, you can do something like this:
```js
if (!this.app.workspace.leftSplit.collapsed) {
 window.open("URL_B");
} else {
 window.open("URL_A");
}
```

~welp

**Reply by FelipeRearden** [reply]
Hello @welpdx !!!!

It's absolutely perfect !!! That's exactly what I needed.

Using `this.app.workspace.leftSplit.collapsed` was a genius idea. It makes the if condition extremely easy to define and understand :)

Thanks again for the rescue ! You are such an  amazing human being!

I wish you a fantastic day ☀️

**Reply by welpdx** [reply]
Happy to help @FelipeRearden! 😊

Let me know if anything else comes up!

**Reply by FelipeRearden** [reply]
> Happy to help @FelipeRearden! 😊

Thank you @welpdx 😀

> Let me know if anything else comes up!

https://github.com/SilentVoid13/Templater/discussions/1050

I am in a rabbit hole with this one for the past 5 hours.  I would appreciate if you could take a look 🙏 ☀️

**Reply by FelipeRearden** [reply]
I dont know how I get this. But I was able to find a solution to this issue 🎉 🎉 🎉 
https://github.com/SilentVoid13/Templater/discussions/1050#discussioncomment-5233261
